src/infra/dex/simulator.rs

```diff
@@ -41,18 +41,26 @@ impl Simulator {
         }
 
         let swapper = contracts::support::Swapper::at(&self.web3, owner);
+        // In the Swapper contract, a dex swap may consist of multiple interactions
```

Comment:
nit: IMO there is no need to have this comment since this is already communicated via the type system / function signature.
```suggestion
```